discord_bots/cogs/economy.py

```diff
@@ -0,0 +1,1119 @@
+            try:
+                await EconomyCommands.create_transaction(
+                    prediction.in_progress_game,
+                    prediction.player,
+                    prediction.prediction_value,
+                    prediction,
+                )
+            except Exception as e:
+                print(f"Exception while refunding predictions for game {game_id}: {e}")
+                raise
+            else:
+                player: Player = (
+                    session.query(Player)
+                    .filter(Player.id == prediction.player_id)
+                    .first()
+                )
+                player.currency += prediction.prediction_value
+                session.commit()
+        session.close()
+
+    async def close_predictions(in_progress_games: list[InProgressGame]):
+        session: SQLAlchemySession = Session()
+        for game in in_progress_games:
+            time_compare: datetime = game.created_at + timedelta(
+                seconds=PREDICTION_TIMEOUT
+            )
+            time_compare = utc.localize(time_compare)
+            if datetime.now(timezone.utc) > time_compare:
+                game.prediction_open = False
+                session.commit()
+        session.close()
+
+    async def create_prediction_message(
+        in_progress_game: InProgressGame, match_channel: TextChannel
+    ) -> int | None:
+        # async def predict(self, interaction: Interaction) -> None:
+
+        if not ECONOMY_ENABLED:
+            return None
+
+        short_game_id: str = in_progress_game.id.split("-")[0]
+        embed = Embed(
+            title=f"Game {short_game_id} Prediction",
+            colour=Colour.blue(),
+        )
+        embed.add_field(
+            name=f"{in_progress_game.team0_name}",
+            value="> Total: 0\n> Win Ratio: 1:1.0\n> Predictors: 0",
+            inline=True,
+        )
+        embed.add_field(
+            name=f"{in_progress_game.team1_name}",
+            value="> Total: 0\n> Win Ratio: 1:1.0\n> Predictors: 0",
+            inline=True,
+        )
+
+        try:
+            prediction_message: Message = await match_channel.send(
+                embed=embed, view=EconomyPredictionView(in_progress_game.id)
+            )
+        except Exception as e:
+            print(f"prediction failed for game: {in_progress_game.id}")
+            return None
+        else:
+            return prediction_message.id
+
+    async def create_prediction(
+        member: Member, game: InProgressGame, selection: int, prediction_value: int
+    ) -> EconomyPrediction:
+        session: SQLAlchemySession = Session()
+        sender: Player = session.query(Player).filter(Player.id == member.id).first()
+
+        if not ECONOMY_ENABLED:
+            raise Exception("Player economy is disabled")
+        elif not sender:
+            raise Exception("Player does not exist")
+        elif not game:
+            raise Exception("Game does not exist")
```

Comment:
The only occasion I can forsee a game not being passed into the `create_prediction`function would be if the modal that calls it somehow got disconnected from the game after the `interaction_check` had already occured, which should not be possible. 

I can remove it, but personally I'd rather keep the edge case exception there just in case.

Reply:
yea but the type being passed in isn't nullable, so this check isn't doing anything

Reply:
Fair point, elif removed.